modules/aws_roles/README.md

```diff
@@ -0,0 +1,177 @@
+
+In order to deploy [ROSA](https://docs.openshift.com/rosa/welcome/index.html) with [STS](https://docs.openshift.com/rosa/rosa_planning/rosa-sts-aws-prereqs.html), AWS Account needs to have the following roles placed:
+
+* Account Roles (One per AWS account)
+* OCM Roles (For OCM UI, One per OCM Org)
+* User Role (For OCM UI, One per OCM user account)
+* Operator Roles (One Per Cluster)
+* OIDC Identity Provider (One Per Cluster)
+
+This terraform module tries to replicate rosa CLI roles creation so that:
+
+* Users have a declartive way to create AWS roles.
+* Users can implement security/infrastructure as code practices.
+* Batch creation of operator roles.
+
+## Prerequisites
+
+* AWS Admin Account
+* OCM Account and OCM CLI
+* ROSA CLI
+
+## Get OCM Information
+
+When creating operator IAM roles, the roles require cluster id, operator role prefix, OIDC endpoint url and thumbprint
+
+
+The information can be retrieved from ocm cli.
+```
+ocm whoami
+{
+  "kind": "Account",
+  "id": "26kcPSEHi0Y6MkTS7OowxfFYmZo",
+  "href": "/api/accounts_mgmt/v1/accounts/26kcPSEHi0Y6MkTS7OowxfFYmZo",
+  "created_at": "2022-03-22T17:43:19Z",
+  "email": "shading+mobb@redhat.com",
+  "first_name": "Shaozhen",
+  "last_name": "Ding",
+  "organization": {
+    "kind": "Organization",
+    "id": "1rkxPO7W12geIcRWITwI0I8VIQV",
+    "href": "/api/accounts_mgmt/v1/organizations/1rkxPO7W12geIcRWITwI0I8VIQV",
+    "created_at": "2021-04-27T14:31:03Z",
+    "ebs_account_id": "7113273",
+    "external_id": "14540493",
+    "name": "Red Hat, Inc.",
+    "updated_at": "2022-06-16T14:17:02Z"
+  },
+  "updated_at": "2022-05-25T02:02:02Z",
+  "username": "shading_mobb"
+}
+```
+
+## Get Clusters Information.
+
+In order to create operator roles for clusters.
+Users need to provide cluster id, OIDC Endpoint URL and thumbprint and operator roles properties list.
+
+```
+ rosa describe cluster -c shaozhenprivate -o json
+{
+  "kind": "Cluster",
+  "id": "1srtno3qggal8ujsegvtb2njvbmhdu8c",
+  "href": "/api/clusters_mgmt/v1/clusters/1srtno3qggal8ujsegvtb2njvbmhdu8c",
+  "aws": {
+    "sts": {
+      "oidc_endpoint_url": "https://rh-oidc.s3.us-east-1.amazonaws.com/1srtno3qggal8ujsegvtb2njvbmhdu8c",
+      "operator_iam_roles": [
+        {
+          "id": "",
+          "name": "ebs-cloud-credentials",
+          "namespace": "openshift-cluster-csi-drivers",
+          "role_arn": "arn:aws:iam::${AWS_ACCOUNT_ID}:role/shaozhenprivate-w4e1-openshift-cluster-csi-drivers-ebs-cloud-cre",
+          "service_account": ""
+        },
+```
+
+In the above example:
+
+* cluster_id =  1srtno3qggal8ujsegvtb2njvbmhdu8c
+* operator_role_prefix = shaozhenprivate-w4e1
+* account_role_prefix = ManagedOpenShift
+* rh_oidc_endpoint_url = rh-oidc.s3.us-east-1.amazonaws.com
+* thumberprint - calculated 
+
+
+The operator roles properties variable is the output of the data source `ocm_rosa_operator_roles` and it's a list of 6 maps which looks like:
+```
+operator_iam_roles = [
```

Comment:
Don't think this is relevant anymore

Reply:
Actually, this is the output of the data source... and it's an example for an input
we must to declare it as an input for the module...